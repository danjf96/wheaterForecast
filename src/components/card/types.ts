import React from "react";
import { ViewStyle } from "react-native";

export interface CARDPROPS {
    style?: ViewStyle,
}