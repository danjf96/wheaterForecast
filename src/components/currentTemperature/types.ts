import { CURRENT } from "../../store/weatherForecast/types";

export interface CURRENTTEMPERATUREPROPS extends CURRENT {
    description: string,
    icon: string,
    windSpeed?: number
}