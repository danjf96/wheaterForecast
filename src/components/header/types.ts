export interface HEADERPROPS {
    text: string
    showIcon?: boolean,
    onPress?: () => void,
    icon?: any
}