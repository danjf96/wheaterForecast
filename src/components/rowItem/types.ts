import { TextStyle, ViewStyle } from "react-native";
import { DAILYPROPS } from "../../store/weatherForecast/types";

export interface ROWITEMPROPS extends DAILYPROPS {
    index?: number
}