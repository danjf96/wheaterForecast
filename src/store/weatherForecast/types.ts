import Geolocation from 'react-native-geolocation-service'
export interface STATEWEATHERFORECAST extends CURREMTWEATHER {
    location?: Geolocation.GeoCoordinates,
    loading: boolean,
    daily?: DAILYPROPS[]
    
}

export interface CURREMTWEATHER {
    base: string,
    clouds: { all: number }
    cod: number,
    coord: { lon: number, lat: number }
    dt: number,
    id: number,
    main: CURRENT,
    name: string,
    sys: { type: number, id: number, country: string, sunrise: number, sunset: number },
    timezone: number,
    visibility: number,
    weather: WHETHER[],
    wind: WIND
}

export interface CURRENT {
    feels_like: number
    humidity: number
    pressure: number
    temp: number
    temp_max: number
    temp_min: number
}

export interface WHETHER {
    description:string
    icon: string
    id: number
    main: string
}

export interface DAILYPROPS {
    clouds: number
    dew_point: number
    dt: number
    feels_like: {day: number, night: number, eve: number, morn: number}
    humidity: number
    moon_phase: number
    moonrise: number
    moonset: number
    pop: number
    pressure: number
    sunrise: number
    sunset: number
    temp: {day: number, min: number, max: number, night: number, eve: number}
    uvi: number
    weather: WHETHER[]
    wind_deg: number
    wind_gust: number
    wind_speed: number
}

export interface WIND { speed: number, deg: number, gust: number}
//TYPES ACTIONS
export const Types = {
    CLEAN: 'CLEAN',
    CHANGE: 'CHANGE_WEATHERFORECAST',
    LOAD: 'LOAD_WEATHERFORECAST',
}