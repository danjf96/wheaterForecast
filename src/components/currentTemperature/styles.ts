import { StyleSheet } from "react-native";
import Colors from "../../assets/Colors";


const Styles = StyleSheet.create({

    container: {
        width: '100%',
        flexDirection: 'column',
        alignItems: 'center'
    },

    containerTwo: { 
        flexDirection: 'row', 
        alignItems: 'center'
    },

    containerMinMax: {
        width: '100%',
        flexDirection: 'column',
        alignItems: 'center',
        marginTop: 15,
    },

    temp: {
        fontSize: 60,
        fontWeight: 'bold',
        color: Colors.text,
        marginBottom: 15
    },

    description: {
        fontSize: 18,
        fontWeight: 'bold',
        textTransform: 'capitalize',
        color: Colors.text
    },

    textWind: {
        fontSize: 20,
        textAlign: 'center',
        marginTop: 15,
        color: Colors.text
    },

    icon: { 
        width: 24, 
        height: 24, 
        marginLeft: 10,
        borderWidth: 2,
        marginRight: 10,
    },

    textMinMax: {
        color: Colors.text,
        marginRight: 10,
        fontSize: 15
    },

    label: {
        color: Colors.text
    }
})

export default Styles