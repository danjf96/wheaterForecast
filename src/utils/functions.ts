export const convertMeterInKM = (value: number | undefined) => {
    //convert meter per second in km per hour
    if(value === undefined) return ''

    return (value * 3.6).toFixed(2)
}