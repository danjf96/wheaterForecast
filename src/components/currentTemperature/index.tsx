import React, { useEffect } from 'react'
import { Text, View } from 'react-native'
import { round } from '../../utils/functions'
import Card from '../card'
import RowItem from '../rowItem'
import Styles from './styles'
import { CURRENTTEMPERATUREPROPS } from './types'

const CurrentTemperature = (props:CURRENTTEMPERATUREPROPS) => {
    const { temp, temp_min, temp_max, description, daily } = props
    return (
        <View style={Styles.container}>

            <View style={Styles.containerTwo}>
                
                <View style={{ flexDirection: 'column', justifyContent: 'center', alignItems: 'center'}}>
                    <Text style={Styles.temp}>{temp?.toPrecision(2)} °C</Text>
                    <Text style={Styles.description}>{description}</Text>
                    <View style={{ flexDirection: 'row', marginTop: 15 }}>
                        <Text style={Styles.textMinMax}>Máx: {round(temp_max)} °C</Text>
                        <Text style={[Styles.textMinMax, { marginRight: 0 }]}>min: {round(temp_min)} °C</Text>
                    </View>

                </View>
            </View>
            


            {daily && <Card>
               {daily?.map( (item, index) => (
                   <RowItem {...item} index={index} key={`item${index}`} />
               ))}
            </Card>}            

        </View>
    )
}

export default CurrentTemperature