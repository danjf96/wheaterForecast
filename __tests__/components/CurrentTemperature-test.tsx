/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import CurrentTemperature from '../../src/components/currentTemperature'
 // Note: test renderer must be required after react-native.
 import { render } from 'react-native-testing-library';
import { convertMeterInKM } from '../../src/utils/functions';
import { CURRENTTEMPERATUREPROPS } from '../../src/components/currentTemperature/types';
 
 const fakeValue:CURRENTTEMPERATUREPROPS = {
    temp: 10, 
    temp_min: 2, 
    temp_max: 11, 
    description: 'testDescription', 
    icon: 'testIcon', 
    windSpeed: 13,
    feels_like: 1,
    humidity: 80,
    pressure: 5
}

it('renders correctly', async () => {
    const screen = render(<CurrentTemperature {...fakeValue}/>)
    expect(screen.toJSON()).toMatchSnapshot()
});

describe('more tests', () => {

    it('values', async () => {
        const { getAllByText } = render(<CurrentTemperature {...fakeValue}/>)
        
        expect(getAllByText(/10 째C|2 째C|11 째C|1 째C|5 mbar|80%|testIcon|testDescription/)).toBeTruthy()
    });

    it('wind convert values', async () => {
        const screen = render(<CurrentTemperature {...fakeValue}/>)
        const wind = convertMeterInKM(fakeValue.windSpeed)
        expect(screen.queryByText(`${wind} Km/h`)).toBeTruthy()
    })
})
 