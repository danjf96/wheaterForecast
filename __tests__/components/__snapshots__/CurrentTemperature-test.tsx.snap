// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 60,
          "fontWeight": "bold",
        }
      }
    >
      10.0
       °C
    </Text>
    <Image
      source={
        Object {
          "uri": "testIcon",
        }
      }
      style={
        Object {
          "height": 48,
          "marginLeft": 10,
          "width": 48,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 25,
        "fontWeight": "bold",
        "textTransform": "capitalize",
      }
    }
  >
    testDescription
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#00AAFF",
          "borderColor": "#0086ff",
          "borderRadius": 15,
          "borderWidth": 1,
          "elevation": 1,
          "marginTop": 25,
          "paddingHorizontal": 20,
          "paddingVertical": 15,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
            undefined,
          ]
        }
      >
        Temp. min:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        2.0 °C
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
            undefined,
          ]
        }
      >
        Temp. max:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        11.0 °C
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
            undefined,
          ]
        }
      >
        Sensação:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        1.0 °C
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
            undefined,
          ]
        }
      >
        Umidade:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        80%
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
            undefined,
          ]
        }
      >
        Pressão:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        5 mbar
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 20,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "600",
            },
            undefined,
          ]
        }
      >
        Vel. do vento:
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 18,
            },
            undefined,
          ]
        }
      >
        46.80 Km/h
      </Text>
    </View>
  </View>
</View>
`;
